crates/services/executor/src/executor.rs

```diff
@@ -1030,228 +1080,671 @@ where
                 gas_price,
                 execution_data,
                 tx_st_transaction,
-                execution_kind,
             ),
         }
     }
 
     #[allow(clippy::too_many_arguments)]
-    fn execute_mint<T>(
+    fn validate_transaction<T>(
```

Comment:
I left comments in other places. Addressing them would help us to use `execute_transaction` instead of `validate_transaction` which should remove redundancy without affecting the business logic or codebase with strange `ExecutionKind` branches=)